### PR TITLE
extract unified RunAuction function to core package

### PR DESCRIPTION
**Pull request by nickpell**
## Extract unified RunAuction function to core package

Refactors auction logic into a reusable `core.RunAuction()` function that handles adjustment factors, floor enforcement, and ranking in a single operation.

### Changes

- Add `core.RunAuction()` function combining adjustment → floor enforcement → ranking
- Add `core.AuctionResult` struct to encapsulate auction outputs (winner, runner-up, eligible/rejected bids)
- Update `enclave/auction.go` to use unified function instead of inline logic
- Add test coverage for RunAuction edge cases

### Why

This ensures TEE and local processing paths use identical auction logic, reducing maintenance burden and preventing behavioral divergence.

### Testing

- All existing tests pass
- Added 6 new test cases covering empty auctions, single bids, floor rejections, and adjustment scenarios

### CODE REVIEW

File: core/auction_test.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	}
+
+	result := RunAuction(bids, nil, nil)
+
+	check.NotNil(t, result)
+	check.NotNil(t, result.Winner)
+
+	// Without adjustments, original ranking is preserved
+	check.Equal(t, "bidder_a", result.Winner.Bidder)
+	check.Equal(t, 2.0, result.Winner.Price)
```
Comment (dsabonis) [inline core/auction_test.go:114]: Should we also validate runner-up?

File: core/auction_test.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	check.NotNil(t, result.Winner)
+	check.NotNil(t, result.RunnerUp)
+
+	// Verify winner (highest bid after adjustment)
+	check.Equal(t, "bidder_a", result.Winner.Bidder)
+	check.Equal(t, 2.0, result.Winner.Price)
+
+	// Verify runner-up
+	check.Equal(t, "bidder_b", result.RunnerUp.Bidder)
+	check.True(t, result.RunnerUp.Price > 1.79 && result.RunnerUp.Price < 1.81)
```
Comment (dsabonis) [inline core/auction_test.go]: @nickpell we only need this tolerance when adjustment factors are non-1.0, right?
I believe we're not using bid-adjustments and don't do currency conversion yet so wonder if we should just remove that functionality? Definitely out of scope for this PR though.

**Review by dsabonis — APPROVED** [review]
LGTM

File: core/auction_test.go (hunk 1 above)
Comment (nickpell) [reply core/auction_test.go:114]: Good idea: 11a71c8

File: core/auction_test.go (hunk 2 above)
Comment (nickpell) [reply core/auction_test.go]: The tolerance check was unnecessary: e47f181
